src/NuGet.Services.Metadata.Catalog.Monitoring/Status/PackageMonitoringStatusService.cs

```diff
@@ -28,40 +26,69 @@ public class PackageMonitoringStatusService : IPackageMonitoringStatusService
         /// </summary>
         private IStorageFactory _storageFactory;
 
-        /// <summary>
-        /// The <see cref="JsonSerializerSettings"/> to use to save and load statuses of packages.
-        /// </summary>
-        private static JsonSerializerSettings SerializerSettings => _serializerSettings.Value;
-        private static Lazy<JsonSerializerSettings> _serializerSettings = new Lazy<JsonSerializerSettings>(() =>
-        {
-            var settings = new JsonSerializerSettings();
-
-            settings.Converters.Add(new NuGetVersionConverter());
-            settings.Converters.Add(new StringEnumConverter());
-
-            return settings;
-        });
-
         public PackageMonitoringStatusService(IStorageFactory storageFactory, ILogger<PackageMonitoringStatusService> logger)
         {
             _logger = logger;
             _storageFactory = storageFactory;
         }
 
-        public async Task<PackageMonitoringStatus> GetAsync(FeedPackageIdentity package, CancellationToken token)
+        public async Task<IEnumerable<PackageMonitoringStatusListItem>> ListAsync(CancellationToken token)
         {
-            var statusTasks =
+            var listTasks =
                 Enum.GetNames(typeof(PackageState))
-                .Select(state =>
-                    Task.Run(async () =>
+                .Select(state => Task.Run(async () =>
+                {
+                    var storage = GetStorage(state);
+                    var list = await storage.List(token);
+                    return list.Select(item =>
                     {
-                        return await GetPackageAsync(GetStorage(state), package, token);
-                    })
-                );
+                        try
+                        {
+                            return new PackageMonitoringStatusListItem(ParsePackageUri(item.Uri), (PackageState)Enum.Parse(typeof(PackageState), state));
+                        }
+                        catch
+                        {
+                            return null;
```

Comment:
Do these exception matter.

Reply:
I don't think they do, but I'll add a log just in case